src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -23,15 +23,18 @@ import java.time.OffsetDateTime
 import java.util.UUID
 
 class Lambda extends RequestStreamHandler {
-
-  val ingestDateTime: OffsetDateTime = OffsetDateTime.now()
-  private lazy val xmlCreator: XMLCreator = XMLCreator(ingestDateTime)
   val dynamoClient: DADynamoDBClient[IO] = DADynamoDBClient[IO]()
   val s3Client: DAS3Client[IO] = DAS3Client[IO]()
 
+  def getXmlCreator: XMLCreator = {
+    val ingestDateTime: OffsetDateTime = OffsetDateTime.now()
+    XMLCreator(ingestDateTime)
+  }
+
   override def handleRequest(inputStream: InputStream, outputStream: OutputStream, context: Context): Unit = {
     val inputString = inputStream.readAllBytes().map(_.toChar).mkString
     val input = read[Input](inputString)
+    val xmlCreator = getXmlCreator
```

Comment:
This is a preference so you can ignore but you could just do `val xmlCreator = XMLCreator(OffsetDateTime.now())` and you wouldn't need the function. I'm not too fussed if it makes it clearer as is though.

Reply:
Ah, yes. The reason, why I'm using a function is so that I can override it in the tests because (for the tests) I need to use an `XMLCreator` with a static datetime.